tests/src/tests/function.spec.ts

```diff
@@ -274,4 +274,33 @@ export = () => {
 		const args: unknown[] = ["123", "456"];
 		expect(select(1, ...args)[1]).to.equal("456");
 	});
+
+	it("should support generator function declarations", () => {
```

Comment:
This title is confusing, because it's not the main point of the test.

I think this fits better in a dedicated file `generator.spec.ts` (Probably along with some more general tests for them)